### DISCUSSION TITLE
SAs still there and DPD useless when VIPs disappear due to interface  reconnect

### DISCUSSION THREAD

**Discussion by p088gll** [strongswan/strongswan#2179] [Q&A] [answered]
We have a road warrior scenario with Ubuntu 22.04 clients configured with swanctl. There is a fortigate on the server side.
Since the network manager plugin from strongswan doesnt support IPv6, we are using charon-systemd. Everything is working fine. The physical interface of the client is getting a virtual IPv4 and IPv6 address from the fortigate, so that it is part of the remote network. 

When the client user for some reason disconnects the phys. interface and reconnects a few seconds afterwards (e.g. by doing a logout and login shortly afterwards), the SAs remain and phase1 communication (DPD) can again take place with the remote fortigate. So for charon, everything is fine. The only problem is that the VIPs are gone on the interface, and they dont come back. So no traffic can go through the tunnels, but the client user is unable to notice it easily. 

Is there a way to recover from this situation (except by manually  restarting charon-systemd altogether) ?

Here is our (anonymized) swanctl config

[list.txt](https://github.com/strongswan/strongswan/files/14704789/list.txt)

**Comment by tobiasbrunner** [comment, marked as answer]
> Since the network manager plugin from strongswan doesnt support IPv6

What do you mean? IPv6 has been supported by that plugin since 5.8.0 (i.e. about five years).

> The only problem is that the VIPs are gone on the interface, and they dont come back.

You might want to install the virtual IPs on a stable interface like `lo` or an XFRM interface (`charon.install_virtual_ip_on`).

**Comment by p088gll** [comment]
Putting the VIPs on the lo Interface solved the problem. Thank you !

Regarding IPv6 support in the Strongswan NM Applet, its good to know that it will come. Ubuntu 22.04 ships with version 5.2.

Actually we like the solution with charon-systemd on the road warrior machines as well. I will open another discussion on this later to keep things seperate.

**Reply by tobiasbrunner** [reply]
> Ubuntu 22.04 ships with version 5.2.

No, Ubuntu 22.04 ships 5.9.5. Even Ubuntu 20.04 ships 5.8.2.

**Reply by p088gll** [reply]
When looking at the nm-appletthe missing IPv6 tab made me assume that it does not support IPv6. Here is a screenshot:
![nm-ipv4](https://github.com/strongswan/strongswan/assets/116447189/99978d3b-498b-4be9-87cb-99b8e9948b05)

Also the list of installed packages is confusing
```
$ dpkg -l | grep strongswan
ii  libstrongswan                                 5.9.5-2ubuntu2.2                        amd64        strongSwan utility and crypto library
ii  libstrongswan-extra-plugins                   5.9.5-2ubuntu2.2                        amd64        strongSwan utility and crypto library (extra plugins)
ii  libstrongswan-standard-plugins                5.9.5-2ubuntu2.2                        amd64        strongSwan utility and crypto library (standard plugins)
ii  network-manager-strongswan                    1.5.2-1                                 amd64        network management framework (strongSwan plugin)
ii  strongswan                                    5.9.5-2ubuntu2.2                        all          IPsec VPN solution metapackage
ii  strongswan-charon                             5.9.5-2ubuntu2.2                        amd64        strongSwan Internet Key Exchange daemon
ii  strongswan-libcharon                          5.9.5-2ubuntu2.2                        amd64        strongSwan charon library
ii  strongswan-nm                                 5.9.5-2ubuntu2.2                        amd64        strongSwan plugin to interact with NetworkManager
ii  strongswan-pki                                5.9.5-2ubuntu2.2                        amd64        strongSwan IPsec client, pki command
ii  strongswan-starter                            5.9.5-2ubuntu2.2                        amd64        strongSwan daemon starter and configuration file parser
ii  strongswan-swanctl                            5.9.5-2ubuntu2.2                        amd64        strongSwan IPsec client, swanctl command
```
I see that the strongswan plugin for nm is 5.9.5 and sure supports IPv6, but maybe the applet itself doesnt ?

**Reply by tobiasbrunner** [reply]
First, these tabs are pretty much useless as basically none of the options are relevant (the only one is search domains to force DNS via VPN, and possibly DNS servers if NM installs those itself). Second, the GNOME applet shows the IPv6 tab since 1.4.5 (Ubuntu ships 1.5.2 of that). However, if you use KDE, then you are not using that but the one for Plasma (might be shipped directly with Plasma's NM frontend), which isn't maintained by us, so no idea what exactly its limitations are (it still uses the same backend, charon-nm, but it might not expose all the supported options).
